### PR TITLE
BaseSpriteController

### PR DESCRIPTION
**Pull request by relinu**
All spriteControllers now derive from BaseSpriteController.
Adds RemoveAll, this lets you delete the visual representation of SpriteControllers

### CODE REVIEW

File: Assets/Scripts/Controllers/FurnitureSpriteController.cs [hunk 1]
```diff
@@ -7,57 +7,97 @@
...
+
+        if (furn.linksToNeighbour == false)
+        {
+            return SpriteManager.current.GetSprite("Furniture", spriteName);
+        }
+
+        // Otherwise, the sprite name is more complicated.
+        spriteName += "_";
+
+        // Check for neighbours North, East, South, West
```
Comment (Randshot) [inline Assets/Scripts/Controllers/FurnitureSpriteController.cs]: Comments should end with a period.

**Comment by Randshot** [comment]
The game runs just fine without any errors.
I have just skimped over the code changes (on GitHub), but from what I have seen it looks pretty clean.

**Comment by Mizipzor** [comment]
This branch has conflicts that must be resolved, probably due to the huge StyleCop cleanup that just happened. Fix them and remove the `[Conflicts]` tag. :)

**Comment by vogonistic** [comment]
Merged!

**Pull request merged**